src/components/ScreenStackItem.tsx

```diff
@@ -95,7 +120,7 @@ function ScreenStackItem(
 
   return (
     <Screen
-      ref={ref}
+      ref={screenRef}
```

Comment:
Instead of the `useLayoutEffect`, you should be able to assign the ref here using a ref callback:

```suggestion
      ref={(node) => {
        screenRef.current = node;
        screenRefs.current[screenId] = node;
      }}
```

Or

```suggestion
      ref={(node) => {
        screenRef.current = node;
        
        if (node == null) {
          delete screenRefs.current[screenId];
        } else {
          screenRefs.current[screenId] = node;
        }
      }}
```

Reply:
I wonder if it's possible that `ref` will be `null`. I like this approach more - it's cleaner, I just wonder if the unmounted views won't stay in `screenRefs`. It shouldn't cause any bugs, but sounds like memory leak.

Reply:
when the component is unmounted, the ref callback will be called with `null`.

Reply:
Changed in: https://github.com/software-mansion/react-native-screens/pull/2476/commits/8d7fddb9eb79c86032d063c190526daa1f29266a